packages/wouter/src/memory-location.js

```diff
@@ -0,0 +1,49 @@
+import mitt from "mitt";
+import { useSyncExternalStore } from "./react-deps.js";
+import { relativePath } from "./paths.js";
+
+/**
+ * In-memory location that supports navigation
+ */
+
+export const memoryLocation = ({
+  path = "/",
+  static: staticLocation,
+  record,
+} = {}) => {
+  let currentPath = path;
+  const history = [currentPath];
+  const emitter = mitt();
+
+  const navigateImplementation = (path, { replace = false } = {}) => {
+    if (record) {
+      if (replace) {
+        history.splice(history.length - 1, 1, path);
+      } else {
+        history.push(path);
+      }
+    }
+
+    currentPath = path;
+    emitter.emit("navigate", path);
+  };
+
+  const navigate = !staticLocation ? navigateImplementation : () => null;
+
+  const subscribe = (cb) => {
+    emitter.on("navigate", cb);
+    return () => emitter.off("navigate", cb);
+  };
+
+  const useMemoryLocation = ({ base } = {}) => {
+    const location = useSyncExternalStore(subscribe, () => currentPath);
+
+    return [relativePath(base, location), navigate];
```

Comment:
I suppose `navigate` method should transform the path provided using `absolutePath` similarly to browser location. How to reproduce:
1. Use base path `"/app"`, initial path `"/app/users"`. The value returned from the hook is `"/users"`.
2. Navigate to `"/settings"`, the path is `"/settings"` (with current setup it will be `"~/settings"`

Reply:
ready!